core/src/main/java/edu/wpi/grip/core/operations/composite/SimpleFilterContoursOperation.java

```diff
@@ -0,0 +1,113 @@
+      OperationDescription.builder()
+          .name("Simple Filter Contours")
+          .summary("Find contours matching certain criteria")
+          .category(OperationDescription.Category.FEATURE_DETECTION)
+          .icon(Icon.iconStream("find-contours"))
+          .build();
+
+  private final SocketHint<ContoursReport> contoursHint = new SocketHint.Builder<>(ContoursReport
+      .class)
+      .identifier("Contours").initialValueSupplier(ContoursReport::new).build();
+
+  private final SocketHint<Number> minAreaHint =
+      SocketHints.Inputs.createNumberSpinnerSocketHint("Min Area", 0, 0, Integer.MAX_VALUE);
+
+  private final SocketHint<Number> minPerimeterHint =
+      SocketHints.Inputs.createNumberSpinnerSocketHint("Min Perimeter", 0, 0, Integer.MAX_VALUE);
+
+
+  private final InputSocket<ContoursReport> contoursSocket;
+  private final InputSocket<Number> minAreaSocket;
+  private final InputSocket<Number> minPerimeterSocket;
+
+  private final OutputSocket<ContoursReport> outputSocket;
+
+  @SuppressWarnings("JavadocMethod")
+  public SimpleFilterContoursOperation(InputSocket.Factory inputSocketFactory, OutputSocket.Factory
+      outputSocketFactory) {
+    this.contoursSocket = inputSocketFactory.create(contoursHint);
+    this.minAreaSocket = inputSocketFactory.create(minAreaHint);
+    this.minPerimeterSocket = inputSocketFactory.create(minPerimeterHint);
+
+    this.outputSocket = outputSocketFactory.create(contoursHint);
+  }
+
+  @Override
+  public List<InputSocket> getInputSockets() {
+    return ImmutableList.of(
+        contoursSocket,
+        minAreaSocket,
+        minPerimeterSocket
+    );
+  }
+
+  @Override
+  public List<OutputSocket> getOutputSockets() {
+    return ImmutableList.of(
+        outputSocket
+    );
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public void perform() {
+    final InputSocket<ContoursReport> inputSocket = contoursSocket;
+    final double minArea = minAreaSocket.getValue().get().doubleValue();
+    final double minPerimeter = minPerimeterSocket.getValue().get().doubleValue();
+
+
+    final MatVector inputContours = inputSocket.getValue().get().getContours();
+    final MatVector outputContours = new MatVector(inputContours.size());
+
+    // Add contours from the input vector to the output vector only if they pass all of the
+    // criteria (minimum area, minimum perimeter)
+    int filteredContourCount = 0;
+    for (int i = 0; i < inputContours.size(); i++) {
+      final Mat contour = inputContours.get(i);
+
+      final double area = contourArea(contour);
+      if (area < minArea) {
+        continue;
+      }
+      if (arcLength(contour, true) < minPerimeter) {
+        continue;
+      }
+      outputContours.put(filteredContourCount++, contour);
+    }
+
+    outputContours.resize(filteredContourCount);
+
+    outputSocket.setValue(new ContoursReport(outputContours,
+        inputSocket.getValue().get().getRows(), inputSocket.getValue().get().getCols()));
```

Comment:
Arguments should all be on one line (if it fits), or each should be on its own line